include/server/api/base_parameters_grammar.hpp

```diff
@@ -111,15 +111,15 @@ struct BaseParametersGrammar : boost::spirit::qi::grammar<Iterator, Signature>
                                                 qi::_1,
                                                 qi::_2)];
 
-        location_rule =
-            (double_ > qi::lit(',') >
-             double_)[qi::_val = ph::bind(
-                          [](double lon, double lat) {
-                              return util::Coordinate(util::toFixed(util::FloatLongitude{lon}),
-                                                      util::toFixed(util::FloatLatitude{lat}));
-                          },
-                          qi::_1,
-                          qi::_2)];
+        location_rule = (double_ > qi::lit(',') >
+                         double_)[qi::_val = ph::bind(
+                                      [](double lon, double lat) {
+                                          return util::Coordinate(
+                                              util::toFixed(util::UnsafeFloatLongitude{lon}),
```

Comment:
https://github.com/Project-OSRM/osrm-backend/blob/safe_coordinate_conversion/unit_tests/util/coordinate_calculation.cpp#L136

Also, we need to catch the user supplying double vales that are large. `toFixed` multiplies the user value by `COORDINATE_PRECISION`, so that can overflow if the user gives a value grater than max(double)/COORDINATE_PRECISION.

So this should remain as I wrote it, right?  Also, this usage is only called once per user coordinate, so doesn't contribute much to total query time. Other usages were called thousands of time per query, they were the real target of this PR.